File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int>ans;
        for(int i=0;i<=n;i++) {
            int c=i,cnt=0;
            while(c!=0) {
                int temp = c&1;
                if(temp) cnt++;
                c=c>>1;
            }
            ans.push_back(cnt);
        }
        return ans;
    }
};

File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int a[] = new int[n+1];
        for(int i=0;i<=n;i++) {
            int c = i,cnt=0;
            while(c!=0) {
                int temp = c&1;
                if(temp==1) cnt++;
                c=c>>1;
            }
            a[i]=cnt;
        }
        return a;
    }
}